packages/contracts/contracts/discovery/Controller.sol

```diff
@@ -162,11 +165,30 @@ contract Controller is IController, AccessControl {
         Batch(batch).setVotingPeriod(start, end);
     }
 
-    function _hasKYC(address _user) internal view returns (bool) {
+    function _hasKYCLiveness(address _user) internal view returns (bool) {
         bytes32 fractalId = FractalRegistry(registry).getFractalId(_user);
         return fractalId != 0;
     }
 
+    function _hasKYCFull(address _user) internal view returns (bool) {
+        bytes32 fractalId = FractalRegistry(registry).getFractalId(_user);
+        return
+            FractalRegistry(registry).isUserInList(fractalId, "plus") &&
+            !_hasKYCBlockedCountry(_user);
+    }
+
+    function _hasKYCBlockedCountry(address _user) internal view returns (bool) {
+        bytes32 fractalId = FractalRegistry(registry).getFractalId(_user);
+        return
+            FractalRegistry(registry).isUserInList(fractalId, "residency_US") ||
+            FractalRegistry(registry).isUserInList(fractalId, "residency_VG") ||
+            FractalRegistry(registry).isUserInList(fractalId, "residency_KY") ||
+            FractalRegistry(registry).isUserInList(fractalId, "residency_KP") ||
+            FractalRegistry(registry).isUserInList(fractalId, "residency_IR") ||
+            FractalRegistry(registry).isUserInList(fractalId, "residency_RU") ||
+            FractalRegistry(registry).isUserInList(fractalId, "residency_VE");
```

Comment:
as discussed on slack, this has 3 major problems:
- it's explicitly PII
- it's a static list of countries, where in reality it will probably need to change over time
- it's not scalable, since for every single user, we need to query for N different lists

all 3 are blockers to this approach, IMO, although the PII thing is definitely the biggest one

Reply:
> as discussed on slack

Can you link me to this pls?

> it's explicitly PII

This was explicitly cleared by our DPO and legal department. The reason is that an EVM address isn't often linkable to a person outside of state-actor forensics. GDPR's issue is not with personal information per se, it's its relation to an identifiable person. If I graffiti your first and last name (which slash much more entropy than your country) on my house in Berlin, I'm not violating the GDPR because nobody has any clue how to associate it with you.

I'm not a fan of this initial approach, but other than principle posturing I can't realistically make a case for why this is actually pragmatically problematic. I'll send you on Slack what I've been thinking about, but I'm not yet convinced of those solution yet either.

> it's a static list of countries, where in reality it will probably need to change over time

Since sales usually covers a short time span, we've rarely seen changes mid-way. Regardless, making the country list in the dependent contact mutable should be a trivial issue for future iterations.

> it's not scalable, since for every single user, we need to query for N different lists

Is this a comment on transaction costs or on the ABI?